### PR TITLE
Set initial files with File or URL string

### LINKED ISSUE #192: set initialFiles with File object array
Right now initialFiles is expecting an array of URLs. I would like to load the dropzone with already uploaded File objects. The initialFiles should be able to accept an array of URLs or File objects.

### PR DESCRIPTION
**Pull request by isaacbuckman**
## Description

- Fixes #192 
- Change initialFiles to take a `File[]`
- Refactor the old initialFiles (taking `string[]`) to initialFileURLs
- Add example to docs of setting initialFiles with simple txt
- All changes made to both DropzoneArea and DropzoneDialog

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - initialFiles now takes `File[]` instead of `string[]`

**Test Configuration**:

- Browser: Chrome

### CODE REVIEW

**Comment by panz3r** [comment]
Hi @isaacbuckman ,

Thanks for your contribution, I'd try to avoid such a breaking change and instead prefer to add a condition on the `initialFiles` prop.

I'd make the prop accept both `string` and `fileObject` and based on the type make the `this.loadInitialFiles` method call `createFileFromUrl(url)` or skip it.

**Comment by isaacbuckman** [comment]
Hey @panz3r,

I agree that the breaking change is probably unnecessary for now. Maybe it is something we can do in the future. I updated the files so the change is no longer breaking.

File: src/components/DropzoneArea.js [hunk 1]
```diff
@@ -49,8 +49,10 @@ class DropzoneArea extends React.PureComponent {
         const {initialFiles} = this.props;
         try {
             const fileObjs = await Promise.all(
-                initialFiles.map(async(url) => {
-                    const file = await createFileFromUrl(url);
+                initialFiles.map(async(file) => {
+                    if (typeof file === 'string' ) {
+                        file = await createFileFromUrl(file);
```
Comment (panz3r) [inline src/components/DropzoneArea.js:54]: Here you should avoid overwriting the `file` parameter.
`file` param name should be `initialFile` and you'd need to add a new temporary variable `file`.

File: src/components/DropzoneArea.js [hunk 2]
```diff
@@ -139,8 +141,13 @@ DropzoneArea.propTypes = {
...
     clearOnUnmount: PropTypes.bool,
-    /** List of URLs of already uploaded images.<br/>**Note:** Please take care of CORS. */
-    initialFiles: PropTypes.arrayOf(PropTypes.string),
+    /** List containing File objects or URL strings.<br/>
+     * **Note:** Please take care of CORS.
+    */
+    initialFiles: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.instanceOf(File)),
+        PropTypes.arrayOf(PropTypes.string),
+    ]),
```
Comment (panz3r) [inline src/components/DropzoneArea.js:150]: The correct typing here is 
```
initialFiles: PropTypes.arrayOf(
  PropTypes.oneOfType([
    PropTypes.instanceOf(File), 
    PropTypes.string
  ])
),
```

File: src/components/DropzoneDialog.js [hunk 3]
```diff
@@ -45,8 +45,10 @@ class DropzoneDialog extends React.PureComponent {
         const {initialFiles} = this.props;
         try {
             const fileObjs = await Promise.all(
-                initialFiles.map(async(url) => {
-                    const file = await createFileFromUrl(url);
+                initialFiles.map(async(file) => {
+                    if (typeof file === 'string' ) {
+                        file = await createFileFromUrl(file);
+                    }
```
Comment (panz3r) [inline src/components/DropzoneDialog.js:51]: See comment above

File: src/components/DropzoneDialog.js [hunk 4]
```diff
@@ -167,8 +169,13 @@ DropzoneDialog.propTypes = {
...
     filesLimit: PropTypes.number,
-    /** List of URLs of already uploaded images.<br/>**Note:** Please take care of CORS. */
-    initialFiles: PropTypes.arrayOf(PropTypes.string),
+    /** List containing File objects or URL strings.<br/>
+     * **Note:** Please take care of CORS.
+    */
+    initialFiles: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.instanceOf(File)),
+        PropTypes.arrayOf(PropTypes.string),
+    ]),
```
Comment (panz3r) [inline src/components/DropzoneDialog.js:178]: See comment above.

File: src/index.d.ts [hunk 5]
```diff
@@ -92,7 +92,7 @@ export type DropzoneAreaProps = Omit<
   'fileObjects' | 'onAdd' | 'onDelete'
 > & {
   clearOnUnmount?: boolean;
-  initialFiles?: string[];
+  initialFiles?: File[] | string[];
```
Comment (panz3r) [inline src/index.d.ts:95]: The correct typing here is 
```ts
initialFiles?: (File | string)[];
```

File: src/index.d.ts [hunk 6]
```diff
@@ -122,7 +122,7 @@ export type DropzoneDialogProps = Omit<
   'fileObjects' | 'onAdd' | 'onDelete' | 'onSave'
 > & {
   clearOnUnmount?: boolean;
-  initialFiles?: string[];
+  initialFiles?: File[] | string[];
```
Comment (panz3r) [inline src/index.d.ts:125]: See comment above.

**Pull request merged**